templates/http/default.conf.j2

```diff
@@ -16,7 +16,11 @@ upstream {{ item.value.upstreams[upstream].name }} {
 
 server {
 {% if item.value.ssl is defined %}
+{% if item.value.port is defined %}
+    listen {{ item.value.port }};
```

Comment:
Can you please do
ssl.port and ssl.opts (that would default just to null-string)? 
especially that they way you do it now, would just open http connection on ssl port (since it lacks ssl parameter).

ssl.opts should default to "ssl" in defaults/main.yml.